src/Sylius/Component/Promotion/Validator/InstructionValidatorInterface.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Lakion package.
+ *
+ * (c) Lakion
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Promotion\Validator;
+
+use Sylius\Component\Promotion\Generator\InstructionInterface;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+interface InstructionValidatorInterface
+{
+    /**
+     * @param InstructionInterface $instruction
+     *
+     * @return bool
+     */
+    public function isGenerationPossible(InstructionInterface $instruction);
```

Comment:
This logic should be split. There should be checker with `isPossible` logic, and  which would use it to build a violation.